internal/manifests/queryfrontend/query_frontend.go

```diff
@@ -253,6 +263,41 @@ func deployment(params manifestutils.Params) (*v1.Deployment, error) {
 	return d, nil
 }
 
+func enableMonitoringTab(tempo v1alpha1.TempoStack, jaegerQueryContainer corev1.Container) (corev1.Container, error) {
+	// TODO (pavolloffay) disable/enable monitoring tab https://github.com/grafana/tempo-operator/issues/464
+	container := corev1.Container{
+		Env: []corev1.EnvVar{
+			{
+				Name:  "METRICS_STORAGE_TYPE",
+				Value: "prometheus",
+			},
+			{
+				Name:  "PROMETHEUS_SERVER_URL",
+				Value: tempo.Spec.Template.QueryFrontend.JaegerQuery.MonitorTab.PrometheusEndpoint,
+			},
+		},
+		Args: []string{
+			"--prometheus.query.support-spanmetrics-connector",
+		},
+	}
+	// If the endpoint matches Prometheus on OpenShift, configure TLS and token based auth
+	prometheusEndpoint := strings.TrimSpace(tempo.Spec.Template.QueryFrontend.JaegerQuery.MonitorTab.PrometheusEndpoint)
+	if prometheusEndpoint == "https://thanos-querier.openshift-monitoring.svc.cluster.local:9091" {
```

Comment:
imho these settings (`prometheus.tls.enabled`, `prometheus.token-file`, `prometheus.tls.ca`) should be part of the `monitorTab` field in the CR, otherwise this feature will only work on non-tls prometheus and the monitoring stack on OpenShift.

Reply:
and to keep it simple for users, set these new fields of the CR in the defaulter webhook to the values here, if the endpoint matches the thanos endpoint